core/src/main/java/smashdudes/ecs/Engine.java

```diff
@@ -41,15 +39,12 @@ public Engine(int worldWidth, int worldHeight, GameOverSystem.IScreenTransition
 
         ExtendViewport viewport = new ExtendViewport(worldWidth,worldHeight, camera);
 
-        RenderDrawSystem rs = new RenderDrawSystem(this, RenderResources.getSpriteBatch());
-        RenderDebugSystem drs = new RenderDebugSystem(this, RenderResources.getShapeRenderer());
-        UIRenderSystem urs = new UIRenderSystem(this, RenderResources.getSpriteBatch(), RenderResources.getFont("KeepOnTruckin", 24));
+        RenderDrawSystem rs = new RenderDrawSystem(this, camera, viewport);
+        RenderDebugSystem drs = new RenderDebugSystem(this, camera, viewport);
 
-        rs.setCamera(camera);
-        drs.setCamera(camera);
-
-        rs.setViewport(viewport);
-        drs.setViewport(viewport);
+        ExtendViewport uiVP = new ExtendViewport(1280, 720);
+        uiVP.update(1280, 720);
```

Comment:
This is needed so that the size of the initial viewport is known to the `UIRenderSystem`.